src/DynamoManipulation/DynamoManipulationExtension.cs

```diff
@@ -174,6 +174,13 @@ private void UnregisterEventHandlers()
                 BackgroundPreviewViewModel.CanNavigateBackgroundPropertyChanged -= Watch3DViewModelNavigateBackgroundPropertyChanged;
                 BackgroundPreviewViewModel.ViewMouseDown -= Watch3DViewModelOnViewMouseDown;
             }
+
+            var settings = GetRunSettings(WorkspaceModel);
+            if (settings != null)
+            {
+                settings.PropertyChanged -= OnRunSettingsPropertyChanged;
```

Comment:
Because when Dynamo is shutdown the setWorkspaceModel call above is not called. 
In a context like Revit where Dynamo can be shutdown and started over and over, this leads to leaks. In Sandbox it's not an issue.